Sources/MongoSwift/ReadWriteConcern.swift

```diff
@@ -114,3 +113,166 @@ extension ReadConcern: Equatable {
+                return lTag == rTag
+            case (.majority, .majority):
+                return true
+            default:
+                return false
+            }
+        }
+    }
+
+    /// Indicates the `W` value for this `WriteConcern`.
+    public var w: W {
+        if let wTag = mongoc_write_concern_get_wtag(self._writeConcern) {
+            return .tag(String(cString: wTag))
+        }
+        let number = mongoc_write_concern_get_w(self._writeConcern)
+        if number == MONGOC_WRITE_CONCERN_W_MAJORITY { return .majority }
+        return .number(number)
+    }
+
+    /// Indicates whether to wait for the write operation to get committed to the journal.
+    public var journal: Bool? {
+        return mongoc_write_concern_get_journal(self._writeConcern)
+    }
+
+    /// If the write concern is not satisfied within this timeout (in milliseconds),
+    /// the operation will return an error. The value MUST be greater than or equal to 0.
+    public var wtimeoutMS: Int32? {
+        return mongoc_write_concern_get_wtimeout(self._writeConcern)
+    }
+
+    /// Indicates whether vbnm this is an acknowledged write concern.
+    public var isAcknowledged: Bool {
+        return mongoc_write_concern_is_acknowledged(self._writeConcern)
+    }
+
+    /// Indicates whether this is the default write concern.
+    public var isDefault: Bool {
+        return mongoc_write_concern_is_default(self._writeConcern)
+    }
+
+    /// Indicates whether the combination of values set on this `WriteConcern` is valid.
+    public var isValid: Bool {
+        return mongoc_write_concern_is_valid(self._writeConcern)
+    }
+
+    /// Initializes a new, empty `WriteConcern`.
+    public init() {
+        self._writeConcern = mongoc_write_concern_new()
+    }
+
+    /// Initializes a new `WriteConcern`.
+    public init(journal: Bool? = nil, w: W? = nil, wtimeoutMS: Int32? = nil) {
+        self._writeConcern = mongoc_write_concern_new()
+        if let journal = journal { mongoc_write_concern_set_journal(self._writeConcern, journal) }
+        if let wtimeoutMS = wtimeoutMS { mongoc_write_concern_set_wtimeout(self._writeConcern, wtimeoutMS) }
+
+        if let w = w {
+            switch w {
+            case let .number(wNumber):
+                mongoc_write_concern_set_w(self._writeConcern, wNumber) 
+            case let .tag(wTag):
+                mongoc_write_concern_set_wtag(self._writeConcern, wTag)
+            case .majority:
+                mongoc_write_concern_set_w(self._writeConcern, MONGOC_WRITE_CONCERN_W_MAJORITY)
+            }
+        }
+    }
+
+    /// Initializes a new `WriteConcern` by copying a `mongoc_write_concern_t`.
+    /// The caller is responsible for freeing the original `mongoc_write_concern_t`.
+    internal init(_ writeConcern: OpaquePointer?) {
+        self._writeConcern = mongoc_write_concern_copy(writeConcern)
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case w, j, wtimeout
```

Comment:
PHPC also doesn't expose the deprecated `fsync` option in our API, but it's still possible for a WriteConcern to have `fsync` set via the libmongoc URI. Is that worth supporting?

If this BSON encoding is used to ferry the WriteConcern's `mongoc_write_concern_t` object into a `bson_t` to pass to a "with_opts" function in libmongoc, you may want to handle it. PHPC is able to use libmongoc's [`mongoc_write_concern_append()`](https://github.com/mongodb/mongo-c-driver/blob/1.10.2/src/libmongoc/src/mongoc/mongoc-write-concern.c#L476) function, which handles `fsync` for us through its internal BSON conversion.

Reply:
I opened [SWIFT-121](https://jira.mongodb.org/browse/SWIFT-121) about this

Reply:
Let's not expose deprecated API unless someone explicitly asks for it